File: colores-api/src/main/java/com/jhosep32dev/coloresapi/model/Color.java
package com.jhosep32dev.coloresapi.model;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.UUID;

@Entity
@Table(name = "colors")
public class Color {

    @Id
    @Column(name = "id")
    private Long id;

    @Column
    @NotNull(message = "Name cannot be null")
    private String name;

    @Column
    @NotNull
    private String color;

    @Column
    private String pantone;

    @Column
    private Integer period;

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public String getPantone() {
        return pantone;
    }

    public void setPantone(String pantone) {
        this.pantone = pantone;
    }

    public Integer getPeriod() {
        return period;
    }

    public void setPeriod(Integer period) {
        this.period = period;
    }
}

File: colores-cli/src/app/model/Color.ts
export class Color {
  private _id: number | null;
  private _name: string | null;
  private _color: string | null;
  private _pantone: string | null;
  private _period: number | null;

  constructor(id: number | null, name: string | null, color: string | null, pantone: string | null, period: number | null) {
    this._id = id;
    this._name = name;
    this._color = color;
    this._pantone = pantone;
    this._period = period;
  }

  get id(): number | null {
    return this._id;
  }

  set id(value: number | null) {
    this._id = value;
  }

  get name(): string | null {
    return this._name;
  }

  set name(value: string | null) {
    this._name = value;
  }

  get color(): string | null {
    return this._color;
  }

  set color(value: string | null) {
    this._color = value;
  }

  get pantone(): string | null {
    return this._pantone;
  }

  set pantone(value: string | null) {
    this._pantone = value;
  }

  get period(): number | null {
    return this._period;
  }

  set period(value: number | null) {
    this._period = value;
  }
}
